components/clp-package-utils/clp_package_utils/scripts/native/dataset_manager.py

```diff
@@ -0,0 +1,259 @@
+import argparse
+import logging
+import shutil
+import sys
+from contextlib import closing
+from pathlib import Path
+from typing import Dict, List
+
+from clp_py_utils.clp_config import ArchiveOutput, Database, StorageType
+from clp_py_utils.clp_metadata_db_utils import (
+    get_archive_tags_table_name,
+    get_archives_table_name,
+    get_column_metadata_table_name,
+    get_datasets_table_name,
+    get_files_table_name,
+    get_tags_table_name,
+)
+from clp_py_utils.s3_utils import s3_delete_by_key_prefix
+from clp_py_utils.sql_adapter import SQL_Adapter
+
+from clp_package_utils.general import (
+    CLP_DEFAULT_CONFIG_FILE_RELATIVE_PATH,
+    CLPConfig,
+    get_clp_home,
+    load_config_file,
+)
+
+# Command/Argument Constants
+LIST_COMMAND: str = "list"
+DEL_COMMAND: str = "del"
+
+logger: logging.Logger = logging.getLogger(__file__)
+
+
+def _handle_list_datasets(datasets: Dict[str, str]) -> int:
+    logger.info(f"Found {len(datasets)} datasets.")
+    for dataset_name, _ in datasets.items():
+        logger.info(dataset_name)
+    return 0
+
+
+def _fetch_existing_datasets_info(
+    db_config: Database,
+) -> Dict[str, str]:
+    sql_adapter = SQL_Adapter(db_config)
+    clp_db_connection_params = db_config.get_clp_connection_params_and_type(True)
+    table_prefix = clp_db_connection_params["table_prefix"]
+    with closing(sql_adapter.create_connection(True)) as db_conn, closing(
+        db_conn.cursor(dictionary=True)
+    ) as db_cursor:
+        db_cursor.execute(
+            f"SELECT name, archive_storage_directory FROM `{get_datasets_table_name(table_prefix)}`"
+        )
+        rows = db_cursor.fetchall()
+        return {row["name"]: row["archive_storage_directory"] for row in rows}
+
+
+def _handle_del_datasets(
+    clp_config: CLPConfig,
+    parsed_args: argparse.Namespace,
+    existing_datasets_info: Dict[str, str],
+):
+    datasets_to_delete: List[str] = []
+    if parsed_args.del_all:
+        datasets_to_delete = list(existing_datasets_info.keys())
+    else:
+        datasets = parsed_args.datasets
+        for dataset in datasets:
+            if dataset not in existing_datasets_info:
+                logger.error(f"Dataset `{dataset}` doesn't exist.")
+                continue
+            datasets_to_delete.append(dataset)
+
+    if 0 == len(datasets_to_delete):
+        logger.warning("No dataset will be deleted...")
```

Comment:
Any suggestion on the warning message?

We may not need this message if we decide to force all input dataset to be valid.